brave/brave-core
# Add built-in support for the wayback machine

close https://github.com/brave/brave-browser/issues/6144

This change adds support for the Wayback Machine by [archive.org](https://archive.org).

**Spec** can be found at https://docs.google.com/document/d/1ioGkGUVmLE0ndfvLRBjDN1d9QtOKeyPOM-33lK0L78k/edit#heading=h.n61l82sr1zxg. For convenience I'm attaching the original infobar below. 

There are two cases for the infobar:

1. When a header response we expect is received:

![Screen Shot 2019-10-30 at 5 37 44 PM](https://user-images.githubusercontent.com/4672033/67897115-833e0480-fb3c-11e9-8347-cfc7e13f3836.png)


1.1 If accepted and Wayback machine has records, redirect the user, close the infobar
1.2 If accepted and Wayback machine has no records, update the infobar, as below:

![Screen Shot 2019-10-30 at 5 37 33 PM](https://user-images.githubusercontent.com/4672033/67897158-94871100-fb3c-11e9-80de-9331f54a08ce.png)

2. If not accepted, close the infobar.

## More snapshots
<img width="1245" alt="Screen Shot 2019-12-10 at 10 50 20 AM" src="https://user-images.githubusercontent.com/6786187/70488523-4baf7880-1b3c-11ea-8fca-c56027505dd8.png">
<img width="838" alt="Screen Shot 2019-12-10 at 10 50 34 AM" src="https://user-images.githubusercontent.com/6786187/70488525-4baf7880-1b3c-11ea-91e0-8fd3aa378068.png">
<img width="844" alt="Screen Shot 2019-12-10 at 10 50 43 AM" src="https://user-images.githubusercontent.com/6786187/70488526-4c480f00-1b3c-11ea-878d-c085dd068b6f.png">
<img width="1205" alt="Screen Shot 2019-12-10 at 10 52 00 AM" src="https://user-images.githubusercontent.com/6786187/70488528-4c480f00-1b3c-11ea-8234-32f3f1c76c43.png">
<img width="1189" alt="Screen Shot 2019-12-10 at 10 52 12 AM" src="https://user-images.githubusercontent.com/6786187/70488529-4c480f00-1b3c-11ea-80dd-bca38235ea40.png">

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan
`npm run test brave_browser_tests -- --filter=BraveWaybackMachineTest.InfobarAddTest`
1. Check wayback option is enabled by default
2. Go to https://brave.com/bo
3. Check wayback infobar is shown
4-1. Click the check button and throbber animation is started
4-1.1 Check an archived version is loaded
4-2. Click x button and infobar is just hidden
5. Go to https://brave.com/abcdef
6. Check wayback infobar is shown
7. Click the check button and infobar is updated with not available message
8. Turn off the wayback option
9. Go to https://brave.com/bo and check wayback infobar is not displayed

## Reviewer Checklist:

- [ ] Request a design review as needed.

## After-merge Checklist:

  changes has landed on.
